paddle/framework/error.h

```diff
@@ -122,15 +117,13 @@ class Error {
    */
   bool isOK() const { return msg_ == nullptr; }
 
-  /**
-   * @brief check this status by glog.
-   * @note It is a temp method used during cleaning Paddle code. It will be
-   *       removed later.
-   */
-  void check() const { CHECK(this->isOK()) << msg(); }
-
-private:
+ private:
   std::shared_ptr<std::string> msg_;
```

Comment:
We should not mainly use `foo(Error* error)` to return Error. `__must_check` will let developer force to check the return error, but `foo(Error* error)` does not have the same technic. So it is better to use `Error __must_check foo();`.

In another hand, [RVO](https://en.wikipedia.org/wiki/Return_value_optimization) depends on compiler implementation.